joomla/cassiopeia
# [WiP] Google Fonts - Generalise option name, clean-up fonts folder and simplify UX

Pull Request for Issue #32 .

### Summary of Changes

- Move the woff and woff2 files to sub-folders, but not the (s)css files, so several (s)css files can share fonts of one or more su-folders.
- More general name of the option, i.e. not use "Google", because fonts could be downloaded and used from any provider, as long as woff and woff2 files are available, and the css could use either one font or a combination of fonts for diverse heading levels and body text.
- Simplify UX: Only one list field for controlling the use of extra fonts, i.e. no need to first click on "Yes" and then select the font, it can be done in one step with a font, and select "None" to switch it off again.
- Remove the description from the remaining field.
- Use a list field and not a filelist field in order to be able to set nice names for the fonts or font combinations. The names don't need to be translated as far as I can see up to now.

### Testing Instructions

1. Checkout the [development branch](https://github.com/joomla/cassiopeia/tree/development) in your Cassipeia local git clone and if necessary pull the latest changes.

2. If never done before:
- Run `composer install` and `npm ci`.
- Install Blog Sample data.

Alternatively to steps 1 and 2 you can make a new installation, using the installation package from the testing PR #48 , [https://test5.richard-fath.de/Joomla_4.0.0-beta4-dev-Cassiopeia-Test-Full_Package_2020-08-30_v2.zip](https://test5.richard-fath.de/Joomla_4.0.0-beta4-dev-Cassiopeia-Test-Full_Package_2020-08-30_v2.zip) , and installing blog sample data.

3. Check content of the `media/fonts` folder.
Result: See 1st screenshot in section "Actual result" below.

4. In backend, check the "Advanced" options of the Cassiopeia template style.
Result: See 2nd screenshot in section "Actual result" below.

5. Apply the changes of this PR.

6. Run `npm ci`.

Alternatively to steps 5 and 6, you can update the installation using the "Upload & Update" tab of the "Joomla Upda" component and using using the installation package from the testing PR #48 : [https://test5.richard-fath.de/Joomla_4.0.0-beta4-dev-Cassiopeia-Test-Update_Package_2020-08-30_v2.zip](https://test5.richard-fath.de/Joomla_4.0.0-beta4-dev-Cassiopeia-Test-Update_Package_2020-08-30_v2.zip).

7. Check content of the `media/fonts` folder.
Result: See 1st screenshot in section "Expected result" below.

8. In backend, check the "Advanced" options of the Cassiopeia template style.
Result: See 2nd screenshot in section "Expected result" below.

9. Toggle the "Extra font(s) ..." dropdown, save settings and check the result in frontend visually and using your browser's inspection tool.
Result: The font selection is applied as well as without this PR.

### Expected result

- The media/fonts folder:
![j4-cassiopeia-pr-50_2](https://user-images.githubusercontent.com/7413183/91663975-58219a00-eaec-11ea-99c9-fcd1d6240a63.png)

- Cassiopeia's template style options:
![j4-cassiopeia-pr-50_1](https://user-images.githubusercontent.com/7413183/91663977-5c4db780-eaec-11ea-9c6a-bb4fd866fd5c.png)

### Actual result

- The media/fonts folder:
![j4-cassiopeia-pr-50_4](https://user-images.githubusercontent.com/7413183/91664081-13e2c980-eaed-11ea-8797-249b053ffbc0.png)

- Cassiopeia's template style options:
![j4-cassiopeia-pr-50_3](https://user-images.githubusercontent.com/7413183/91663979-67084c80-eaec-11ea-8652-18660005a9f9.png)

### Documentation Changes Required

None beside what has to be done for the new option anyway.
